app/src/lib/stores/pull-request-store.ts

```diff
@@ -25,207 +25,264 @@ import { IRemote } from '../../models/remote'
   ): ReadonlyArray<IRemote> {
     const forkedRemotes = remotes.filter(remote =>
       remote.name.startsWith(ForkedRemotePrefix)
     )
     const remotesOfPullRequests = new Set<string>()
-    openPullRequests.forEach(openPullRequest => {
-      const { gitHubRepository } = openPullRequest.head
+
+    openPullRequests.forEach(pr => {
+      const { gitHubRepository } = pr.head
+
       if (gitHubRepository != null && gitHubRepository.cloneURL != null) {
         remotesOfPullRequests.add(gitHubRepository.cloneURL)
       }
     })
-    const forkedRemotesToDelete = forkedRemotes.filter(
+
+    const result = forkedRemotes.filter(
       forkedRemote => !remotesOfPullRequests.has(forkedRemote.url)
     )
 
-    return forkedRemotesToDelete
+    return result
   }
 
-  private async deleteForkedRemotes(
+  private async deleteRemotes(
     repository: Repository,
     remotes: ReadonlyArray<IRemote>
   ) {
-    for (const remote of remotes) {
-      await removeRemote(repository, remote.name)
-    }
+    const promises: Array<Promise<void>> = []
+
+    remotes.forEach(r => promises.push(removeRemote(repository, r.name)))
+    await Promise.all(promises)
   }
 
-  private changeActiveFetchCount(
+  private updateActiveFetchCount(
     repository: GitHubRepository,
-    fn: (count: number) => number
+    update: (count: number) => number
   ) {
-    const key = forceUnwrap(
+    const repoDbId = forceUnwrap(
       'Cannot fetch PRs for a repository which is not in the database',
       repository.dbID
     )
-    const currentCount = this.activeFetchCountPerRepository.get(key) || 0
-    const newCount = fn(currentCount)
-    this.activeFetchCountPerRepository.set(key, newCount)
+    const currentCount = this._activeFetchCountPerRepository.get(repoDbId) || 0
+    const newCount = update(currentCount)
 
+    this._activeFetchCountPerRepository.set(repoDbId, newCount)
     this.emitUpdate(repository)
   }
 
-  /** Is the store currently fetching the list of open pull requests? */
-  public isFetchingPullRequests(repository: GitHubRepository): boolean {
-    const key = forceUnwrap(
-      'Cannot fetch PRs for a repository which is not in the database',
-      repository.dbID
-    )
-
-    const currentCount = this.activeFetchCountPerRepository.get(key) || 0
-    return currentCount > 0
-  }
-
-  /** Loads the status for the given pull request. */
-  public async refreshSinglePullRequestStatus(
-    repository: GitHubRepository,
-    account: Account,
-    pullRequest: PullRequest
-  ): Promise<void> {
-    await this.refreshStatusForPRs([pullRequest], repository, account)
-  }
-
-  /** Loads the status for all pull request against a given repository. */
-  public async refreshPullRequestStatuses(
-    repository: GitHubRepository,
-    account: Account
-  ): Promise<void> {
-    const prs = await this.getPullRequests(repository)
-
-    await this.refreshStatusForPRs(prs, repository, account)
-  }
-
   private async refreshStatusForPRs(
```

Comment:
Similar thing here. `refresh` is a bit unclear - if we want to make this a bit more data- and database-focused we should also name this `fetchAndCacheStatusesForPRs`